diff --git a/node_modules/expo-share-intent/plugin/build/ios/withIosShareExtensionXcodeTarget.js b/node_modules/expo-share-intent/plugin/build/ios/withIosShareExtensionXcodeTarget.js
index c9cbd8b..734393b 100644
--- a/node_modules/expo-share-intent/plugin/build/ios/withIosShareExtensionXcodeTarget.js
+++ b/node_modules/expo-share-intent/plugin/build/ios/withIosShareExtensionXcodeTarget.js
@@ -71,6 +71,12 @@ const withShareExtensionXcodeTarget = (config, parameters) => {
                     buildSettingsObj["SWIFT_EMIT_LOC_STRINGS"] = "YES";
                     buildSettingsObj["SWIFT_VERSION"] = "5.0";
                     buildSettingsObj["TARGETED_DEVICE_FAMILY"] = `"1,2"`;
+
+                    // Set the development team ID if provided
+                    // This is useful for automatic signing in Xcode
+                    if (parameters.developmentTeamId) {
+						buildSettingsObj["DEVELOPMENT_TEAM"] = `"${parameters.developmentTeamId}"`
+					}
                 }
             }
         }
