diff --git a/node_modules/react-native-ios-context-menu/src/components/ContextMenuView/ContextMenuView.tsx b/node_modules/react-native-ios-context-menu/src/components/ContextMenuView/ContextMenuView.tsx
index f3dddf3..b14a7d3 100644
--- a/node_modules/react-native-ios-context-menu/src/components/ContextMenuView/ContextMenuView.tsx
+++ b/node_modules/react-native-ios-context-menu/src/components/ContextMenuView/ContextMenuView.tsx
@@ -367,7 +367,7 @@ export class ContextMenuView extends
               nativeID={NATIVE_ID_KEYS.detachedView}
               shouldImmediatelyDetach={true}
             >
-              <React.Fragment>
+              <View>
                 {isUsingCustomPreview && (
                   <RNIDetachedViewContent
                     nativeID={NATIVE_ID_KEYS.contextMenuPreview}
@@ -377,8 +377,8 @@ export class ContextMenuView extends
                     }
                   </RNIDetachedViewContent>
                 )}
-              </React.Fragment>
-              <React.Fragment>
+              </View>
+              <View>
                 {isUsingAuxillaryPreview && (
                   <RNIDetachedViewContent
                     nativeID={NATIVE_ID_KEYS.contextMenuAuxiliaryPreview}
@@ -388,7 +388,7 @@ export class ContextMenuView extends
                     }
                   </RNIDetachedViewContent>
                 )}
-              </React.Fragment>
+              </View>
             </RNIDetachedView>
           )}
           <View>
