diff --git a/node_modules/react-native-saf-x/android/src/main/java/com/reactnativesafx/utils/DocumentStat.java b/node_modules/react-native-saf-x/android/src/main/java/com/reactnativesafx/utils/DocumentStat.java
index 59e0b5f..bdf60a8 100644
--- a/node_modules/react-native-saf-x/android/src/main/java/com/reactnativesafx/utils/DocumentStat.java
+++ b/node_modules/react-native-saf-x/android/src/main/java/com/reactnativesafx/utils/DocumentStat.java
@@ -52,8 +52,20 @@ public class DocumentStat {
     final int mimeTypeColIndex = c.getColumnIndex(DocumentsContract.Document.COLUMN_MIME_TYPE);
     final int sizeColIndex = c.getColumnIndex(DocumentsContract.Document.COLUMN_SIZE);
     final int lastModifiedColIndex = c.getColumnIndex(DocumentsContract.Document.COLUMN_LAST_MODIFIED);
+    final int displayNameColIndex = c.getColumnIndex(DocumentsContract.Document.COLUMN_DISPLAY_NAME);
+    final int documentIdColIndex = c.getColumnIndex(DocumentsContract.Document.COLUMN_DOCUMENT_ID);
 
-    this.displayName = c.getString(1);
+    //this.displayName = c.getString(1);
+
+    if (displayNameColIndex != -1) {
+      this.displayName = c.getString(displayNameColIndex);
+    } else {
+      if (documentIdColIndex != -1) {
+        this.displayName = c.getString(documentIdColIndex);
+      } else {
+        this.displayName = "Unknown file";
+      }
+    }
 
     if (mimeTypeColIndex != -1) {
       this.mimeType = c.getString(mimeTypeColIndex);