diff --git a/node_modules/argon2/binding.gyp b/node_modules/argon2/binding.gyp
index d880b52..18507e7 100644
--- a/node_modules/argon2/binding.gyp
+++ b/node_modules/argon2/binding.gyp
@@ -1,4 +1,7 @@
 {
+  "variables": {
+    "iossim%": "false"
+  },
   "target_defaults": {
     "include_dirs": ["argon2/include"],
     "target_conditions": [
@@ -9,6 +12,36 @@
           "MACOSX_DEPLOYMENT_TARGET": "10.7"
         }
       }],
+      ["OS == 'ios'", {
+        "xcode_settings": {
+          "CLANG_CXX_LIBRARY": "libc++",
+          "GCC_ENABLE_CPP_EXCEPTIONS": "YES",
+          "IPHONEOS_DEPLOYMENT_TARGET": "9.0",
+          "TARGETED_DEVICE_FAMILY": "1,2",
+          "ONLY_ACTIVE_ARCH": "NO",
+          "CODE_SIGNING_ALLOWED": "NO",
+          "CODE_SIGNING_REQUIRED": "NO",
+          "ENABLE_BITCODE": "YES"
+        }
+      }],
+      ["OS == 'android'", {
+        "cflags": ["-fPIC"],
+        "cflags_cc": ["-fPIC"],
+        "conditions": [
+          ["target_arch == 'arm'", {
+            "cflags": ["-march=armv7-a", "-mfloat-abi=softfp", "-mfpu=neon"],
+            "ldflags": ["-march=armv7-a"]
+          }],
+          ["target_arch == 'arm64'", {
+            "cflags": ["-march=armv8-a"],
+            "ldflags": ["-march=armv8-a"]
+          }],
+          ["target_arch == 'x64'", {
+            "cflags": ["-m64"],
+            "ldflags": ["-m64"]
+          }]
+        ]
+      }],
       ["OS == 'win'", {
         "defines+": ["_HAS_EXCEPTIONS=1"],
         "msvs_settings": {
@@ -20,13 +53,9 @@
       "Release": {
         "target_conditions": [
           ["OS not in 'ios mac'", {
-            # Avoid defining _FORTIFY_SOURCE on Darwin
             "defines+": ["_FORTIFY_SOURCE=2"]
           }],
           ["OS not in 'win ios mac aix'", {
-            # On Darwin with Xcode CLT/LLVM, "-fvisibility=hidden" hide all symbols that
-            # not explicitly marked with __attribute__((visibility("default")))
-            # Flags for sections are specific to ELF binaries
             "cflags+": ["-fdata-sections", "-ffunction-sections", "-fvisibility=hidden"],
             "ldflags+": ["-Wl,--gc-sections"]
           }]
@@ -47,15 +76,39 @@
       ],
       "cflags+": ["-Wno-type-limits"],
       "conditions": [
-        ["target_arch == 'ia32' or target_arch == 'x64'", {
-          "cflags+": ["-msse", "-msse2"],
-          "sources+": ["argon2/src/opt.c"]
-        }, {
-          "sources+": ["argon2/src/ref.c"]
+        ["OS == 'ios'", {
+          "type": "static_library",
+          "sources+": ["argon2/src/ref.c"],
+          "xcode_settings": {
+            "OTHER_CFLAGS": ["-fembed-bitcode-marker"]
+          }
+        }],
+        ["OS == 'android'", {
+          "type": "static_library",
+          "conditions": [
+            ["target_arch in 'arm arm64'", {
+              "sources+": ["argon2/src/ref.c"]
+            }],
+            ["target_arch == 'x64'", {
+              "cflags+": ["-msse", "-msse2"],
+              "sources+": ["argon2/src/opt.c"]
+            }]
+          ]
+        }],
+        ["OS not in 'ios android'", {
+          "conditions": [
+            ["target_arch == 'ia32' or target_arch == 'x64'", {
+              "cflags+": ["-msse", "-msse2"],
+              "sources+": ["argon2/src/opt.c"]
+            }, {
+              "sources+": ["argon2/src/ref.c"]
+            }]
+          ],
+          "type": "static_library"
         }]
-      ],
-      "type": "static_library"
-    }, {
+      ]
+    }, 
+    {
       "target_name": "argon2",
       "defines+": [
         "NAPI_VERSION=<(napi_build_version)",
@@ -65,10 +118,26 @@
       "sources": [
         "argon2.cpp"
       ],
-      "cflags_cc+": ["-Wall", "-Wextra", "-Wformat", "-Wnon-virtual-dtor", "-pedantic", "-Werror", "-fexceptions"],
-      "cflags_cc!": ["-fno-exceptions"],
       "include_dirs": ["<!(node -p \"require('node-addon-api').include_dir\")"],
       "dependencies": ["libargon2"],
+      "conditions": [
+        ["OS == 'ios'", {
+          "cflags_cc+": ["-Wall", "-Wextra", "-Wno-unused-parameter", "-fexceptions"],
+          "xcode_settings": {
+            "GCC_ENABLE_CPP_EXCEPTIONS": "YES",
+            "OTHER_CPLUSPLUSFLAGS": ["-fembed-bitcode-marker"]
+          }
+        }],
+        ["OS == 'android'", {
+          "cflags_cc+": ["-Wall", "-Wextra", "-Wno-unused-parameter", "-fexceptions"],
+          "cflags_cc!": ["-fno-exceptions", "-fno-rtti"],
+          "cflags!": ["-fno-exceptions", "-fno-rtti"]
+        }],
+        ["OS not in 'ios android'", {
+          "cflags_cc+": ["-Wall", "-Wextra", "-Wformat", "-Wnon-virtual-dtor", "-pedantic", "-Werror", "-fexceptions"],
+          "cflags_cc!": ["-fno-exceptions"]
+        }]
+      ],
       "configurations": {
         "Debug": {
           "conditions": [
diff --git a/node_modules/argon2/prebuilds/darwin-arm64/argon2.armv8.glibc.node b/node_modules/argon2/prebuilds/darwin-arm64/argon2.armv8.glibc.node
deleted file mode 100644
index 394d311..0000000
Binary files a/node_modules/argon2/prebuilds/darwin-arm64/argon2.armv8.glibc.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/darwin-x64/argon2.glibc.node b/node_modules/argon2/prebuilds/darwin-x64/argon2.glibc.node
deleted file mode 100644
index dbe941a..0000000
Binary files a/node_modules/argon2/prebuilds/darwin-x64/argon2.glibc.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/freebsd-x64/argon2.glibc.node b/node_modules/argon2/prebuilds/freebsd-x64/argon2.glibc.node
deleted file mode 100644
index 4e745ab..0000000
Binary files a/node_modules/argon2/prebuilds/freebsd-x64/argon2.glibc.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/linux-arm/argon2.armv7.glibc.node b/node_modules/argon2/prebuilds/linux-arm/argon2.armv7.glibc.node
deleted file mode 100644
index 552a5f9..0000000
Binary files a/node_modules/argon2/prebuilds/linux-arm/argon2.armv7.glibc.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/linux-arm/argon2.armv7.musl.node b/node_modules/argon2/prebuilds/linux-arm/argon2.armv7.musl.node
deleted file mode 100644
index cc9210a..0000000
Binary files a/node_modules/argon2/prebuilds/linux-arm/argon2.armv7.musl.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/linux-arm64/argon2.armv8.glibc.node b/node_modules/argon2/prebuilds/linux-arm64/argon2.armv8.glibc.node
deleted file mode 100644
index 9dd25a1..0000000
Binary files a/node_modules/argon2/prebuilds/linux-arm64/argon2.armv8.glibc.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/linux-arm64/argon2.armv8.musl.node b/node_modules/argon2/prebuilds/linux-arm64/argon2.armv8.musl.node
deleted file mode 100644
index 2f1ee15..0000000
Binary files a/node_modules/argon2/prebuilds/linux-arm64/argon2.armv8.musl.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/linux-x64/argon2.glibc.node b/node_modules/argon2/prebuilds/linux-x64/argon2.glibc.node
deleted file mode 100644
index cff560c..0000000
Binary files a/node_modules/argon2/prebuilds/linux-x64/argon2.glibc.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/linux-x64/argon2.musl.node b/node_modules/argon2/prebuilds/linux-x64/argon2.musl.node
deleted file mode 100644
index ac7abf2..0000000
Binary files a/node_modules/argon2/prebuilds/linux-x64/argon2.musl.node and /dev/null differ
diff --git a/node_modules/argon2/prebuilds/win32-x64/argon2.glibc.node b/node_modules/argon2/prebuilds/win32-x64/argon2.glibc.node
deleted file mode 100644
index 2c9ac64..0000000
Binary files a/node_modules/argon2/prebuilds/win32-x64/argon2.glibc.node and /dev/null differ
